### PR TITLE
Fix being able to shrink guns with defib chargers

### LINKED ISSUE #6685: Egun can go into the defib charger, and when it comes out it's tiny.
(WEB REPORT BY: cisrox REMOTE: 158.69.55.2:7777)
>Revision 3bbf052ed0e3df72cb97a686f2414a4142bbf829

>General description of the issue
You can put eguns in the defib charger, and they come out small.

>What you expected to happen
I expected it to not go in at all, and at the least not come out tiny.

>What actually happened
It turned small, and also got charged

>Steps to reproduce if possible

Put e-gun in the defib charger. Take out and it turns into a shrimp.

### PR DESCRIPTION
**Pull request by ghost**
It was either not returning once we were sure we weren't dealing with a defibrillator charger, or the parent condition call

I did both, it works correctly now

Fixes #6685

### CODE REVIEW

**Comment by ghost** [comment]
:-1: this sounds like a hilarious bug

**Comment by ghost** [comment]
It's going to get a lot less hilarious when the players will find out and abuse it constantly

If you really want this to be permanent behavior, create a gun-shrinking admin-only entity that transforms the gun permanently

**Comment by Exxion** [comment]
Come on, just fix it so it only works once
In that case it isn't even a balance issue because defib chargers don't charge guns (I think)

File: code/game/machinery/defibcharger.dm [hunk 1]
```diff
@@ -87,25 +87,24 @@ obj/machinery/recharger/defibcharger/wallcharger/process()
 	return -1
 
 obj/machinery/recharger/defibcharger/wallcharger/attackby(obj/item/weapon/G as obj, mob/user as mob)
-	if(..())
```
Comment (clusterfack) [inline code/game/machinery/defibcharger.dm]: As a child of machinery this has many more consequences to being removed than is apparent

**Comment by clusterfack** [comment]
With the current fix cant you still put guns in the defib charger? Because that needs to be prevented as well.

**Comment by d3athrow** [comment]
Yeah no reason to be able to put them in the charger period.

**Comment by ghost** [comment]
I need to check if that's working as intended once I finally get spare time

**Comment by ghost** [comment]
t. what the fuck am I doing anyways

**Pull request closed without merging**